### PR TITLE
Check Transaction support

### PR DESCRIPTION
**Pull request by florent-uzio**
# Title

This PR aims to add the relevant functions for the Check transactions (Create, Cash and Cancel).
The tests cover 100% of the transactions.

### CODE REVIEW

File: xrpl/transaction/check_cancel.go [hunk 1]
```diff
@@ -1,19 +1,57 @@
...
+}
+
+// Validate checks the validity of the CheckCancel transaction.
+func (c *CheckCancel) Validate() (bool, error) {
+	ok, err := c.BaseTx.Validate()
+	if err != nil || !ok {
+		return false, err
+	}
+
+	if !typecheck.IsHex(c.CheckID.String()) {
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/check_cancel.go:52]: Check string length is 64 characters

File: xrpl/transaction/check_create.go [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
-	InvoiceID      types.Hash256 `json:",omitempty"`
+	// The unique address of the account that can cash the Check.
+	Destination types.Address
+	// Maximum amount of source currency the Check is allowed to debit the sender, including transfer fees on non-XRP currencies.
+	// The Check can only credit the destination with the same currency (from the same issuer, for non-XRP currencies). For non-XRP amounts, the nested field names MUST be lower-case.
+	SendMax types.CurrencyAmount
+	// (Optional) Arbitrary tag that identifies the reason for the Check, or a hosted recipient to pay.
+	DestinationTag uint32 `json:",omitempty"`
+	// (Optional) Time after which the Check is no longer valid, in seconds since the Ripple Epoch.
+	Expiration uint32 `json:",omitempty"`
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/check_create.go:37]: Add 0 by default

**Review by GuillemGarciaDev — CHANGES_REQUESTED** [review]

File: xrpl/transaction/check_create.go (hunk 2 above)
Comment (florent-uzio) [reply xrpl/transaction/check_create.go:37]: I am not sure it's a good idea to set it to 0 by default as it's an optional field. It's possible to define the CheckCreate without it and setting it to 0 by default might confuse the user using the SDK.
What do you think @GuillemGarciaDev ?

Comment (GuillemGarciaDev) [reply xrpl/transaction/check_create.go:37]: It is set by default if an integer variable is not initialized with a given value. It is a go language feature.

**Review by GuillemGarciaDev — APPROVED** [review]

**Pull request merged**